dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/AuthorityEntryLinkRepository.java

```diff
@@ -50,12 +51,12 @@
     private AuthorityUtils authorityUtils;
 
     @PreAuthorize("hasAuthority('AUTHENTICATED')")
-    public Page<AuthorityEntryRest> query(HttpServletRequest request, String name,
-                                          Pageable pageable, Projection projection) {
+    public Page<AuthorityEntryRest> query(@Nullable HttpServletRequest request, String name,
```

Comment:
What's the use case we need AuthorityEntryLinkRepository to have a nullable request and pageable?
This change was not clear to me

Reply:
The reason I added these indicators was mainly because there's an implicit contract in link repository getter methods, and I wanted the code to be as explicit (as possible) about that. In general, any time a link repository is called while embedding a subresource into another resource, those can be null.

I tried to resist refactoring too much in these early PRs, but I would like to eventually see some refactoring with link repository methods in general. I think they could adhere to a real java interface (maybe one for paged subresources, and one for scalar ones), in which case, if it was really important to pass down something like the HttpServletRequest, the fact that it could be nullable would be part of the java interface, not necessarily the impl.